api/client/start.go

```diff
@@ -112,7 +112,7 @@ func (cli *DockerCli) CmdStart(args ...string) error {
 		})
 
 		// 3. Start the container.
-		if err := cli.client.ContainerStart(context.Background(), container); err != nil {
+		if err := cli.client.ContainerStart(context.Background(), container, ""); err != nil {
```

Comment:
Allocate the context at the top of the function. It is useless if it is not shared (see above: a cancel gets completely ignored here.